assert.NotSubset incorrecly formats with %q

## Description
NotSubset (but not Subset) seems to be using %q to format error output, that doesn't work for many collection element types.

## Step To Reproduce
```go
package kata_test

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

type User struct {
	ID, Name string
	Age      int
}

func TestIfy(t *testing.T) {
	assert.NotSubset(t, []bool{true}, []bool{true})
}
```

## Expected behavior
Values of arguments to be formatted well.

## Actual behavior
```
--- FAIL: TestIfy (0.00s)
    /Users/brackendawson/src/github.com/brackendawson/kata/kata_test.go:15: 
        	Error Trace:	/Users/brackendawson/src/github.com/brackendawson/kata/kata_test.go:15
        	Error:      	[%!q(bool=true)] is a subset of [%!q(bool=true)]
        	Test:       	TestIfy
FAIL
coverage: 0.0% of statements
FAIL	github.com/brackendawson/kata	0.202s
FAIL
```

--- a/assert/assertions.go
+++ b/assert/assertions.go
@@ -1083,12 +1083,12 @@ func NotSubset(t TestingT, list, subset interface{}, msgAndArgs ...interface{}) 
 
 	listKind := reflect.TypeOf(list).Kind()
 	if listKind != reflect.Array && listKind != reflect.Slice && listKind != reflect.Map {
-		return Fail(t, fmt.Sprintf("%q has an unsupported type %s", list, listKind), msgAndArgs...)
+		return Fail(t, fmt.Sprintf("%#v has an unsupported type %s", list, listKind), msgAndArgs...)
 	}
 
 	subsetKind := reflect.TypeOf(subset).Kind()
 	if subsetKind != reflect.Array && subsetKind != reflect.Slice && subsetKind != reflect.Map {
-		return Fail(t, fmt.Sprintf("%q has an unsupported type %s", subset, subsetKind), msgAndArgs...)
+		return Fail(t, fmt.Sprintf("%#v has an unsupported type %s", subset, subsetKind), msgAndArgs...)
 	}
 
 	if subsetKind == reflect.Map && listKind == reflect.Map {
@@ -1107,7 +1107,7 @@ func NotSubset(t TestingT, list, subset interface{}, msgAndArgs ...interface{}) 
 			}
 		}
 
-		return Fail(t, fmt.Sprintf("%s is a subset of %s", truncatingFormat("%q", subset), truncatingFormat("%q", list)), msgAndArgs...)
+		return Fail(t, fmt.Sprintf("%s is a subset of %s", truncatingFormat("%#v", subset), truncatingFormat("%#v", list)), msgAndArgs...)
 	}
 
 	subsetList := reflect.ValueOf(subset)
@@ -1122,14 +1122,14 @@ func NotSubset(t TestingT, list, subset interface{}, msgAndArgs ...interface{}) 
 		element := subsetList.Index(i).Interface()
 		ok, found := containsElement(list, element)
 		if !ok {
-			return Fail(t, fmt.Sprintf("%q could not be applied builtin len()", list), msgAndArgs...)
+			return Fail(t, fmt.Sprintf("%#v could not be applied builtin len()", list), msgAndArgs...)
 		}
 		if !found {
 			return true
 		}
 	}
 
-	return Fail(t, fmt.Sprintf("%s is a subset of %s", truncatingFormat("%q", subset), truncatingFormat("%q", list)), msgAndArgs...)
+	return Fail(t, fmt.Sprintf("%s is a subset of %s", truncatingFormat("%#v", subset), truncatingFormat("%#v", list)), msgAndArgs...)
 }
 
 // ElementsMatch asserts that the specified listA(array, slice...) is equal to specified

--- a/assert/assertions_test.go
+++ b/assert/assertions_test.go
@@ -1202,10 +1202,10 @@ func TestSubsetNotSubset(t *testing.T) {
 	}{
 		// cases that are expected to contain
 		{[]int{1, 2, 3}, nil, true, `nil is the empty set which is a subset of every set`},
-		{[]int{1, 2, 3}, []int{}, true, `[] is a subset of ['\x01' '\x02' '\x03']`},
-		{[]int{1, 2, 3}, []int{1, 2}, true, `['\x01' '\x02'] is a subset of ['\x01' '\x02' '\x03']`},
-		{[]int{1, 2, 3}, []int{1, 2, 3}, true, `['\x01' '\x02' '\x03'] is a subset of ['\x01' '\x02' '\x03']`},
-		{[]string{"hello", "world"}, []string{"hello"}, true, `["hello"] is a subset of ["hello" "world"]`},
+		{[]int{1, 2, 3}, []int{}, true, `[]int{} is a subset of []int{1, 2, 3}`},
+		{[]int{1, 2, 3}, []int{1, 2}, true, `[]int{1, 2} is a subset of []int{1, 2, 3}`},
+		{[]int{1, 2, 3}, []int{1, 2, 3}, true, `[]int{1, 2, 3} is a subset of []int{1, 2, 3}`},
+		{[]string{"hello", "world"}, []string{"hello"}, true, `[]string{"hello"} is a subset of []string{"hello", "world"}`},
 		{map[string]string{
 			"a": "x",
 			"c": "z",
@@ -1213,8 +1213,8 @@ func TestSubsetNotSubset(t *testing.T) {
 		}, map[string]string{
 			"a": "x",
 			"b": "y",
-		}, true, `map["a":"x" "b":"y"] is a subset of map["a":"x" "b":"y" "c":"z"]`},
-		{[]string{"a", "b", "c"}, map[string]int{"a": 1, "c": 3}, true, `map["a":'\x01' "c":'\x03'] is a subset of ["a" "b" "c"]`},
+		}, true, `map[string]string{"a":"x", "b":"y"} is a subset of map[string]string{"a":"x", "b":"y", "c":"z"}`},
+		{[]string{"a", "b", "c"}, map[string]int{"a": 1, "c": 3}, true, `map[string]int{"a":1, "c":3} is a subset of []string{"a", "b", "c"}`},
 
 		// cases that are expected not to contain
 		{[]string{"hello", "world"}, []string{"hello", "testify"}, false, `[]string{"hello", "world"} does not contain "testify"`},
@@ -4025,15 +4025,24 @@ func TestSubsetWithMapTooLongToPrint(t *testing.T) {
 	Contains(t, mockT.errorString(), `<... truncated> does not contain map[bool][]int{false:[]int{0, 0, 0,`)
 }
 
+// Regression test for https://github.com/stretchr/testify/issues/1800:
+// NotSubset was using %q to format error output, producing garbage like
+// [%!q(bool=true)] for non-string element types.
+func TestNotSubsetFormatsNonStringElementsCorrectly(t *testing.T) {
+	mockT := new(mockTestingT)
+	NotSubset(mockT, []bool{true}, []bool{true})
+	errStr := mockT.errorString()
+	NotContains(t, errStr, "%!q", "NotSubset error message should not contain %%q formatting artifacts")
+	Contains(t, errStr, "true", "NotSubset error message should contain the formatted value")
+}
+
 func TestNotSubsetWithSliceTooLongToPrint(t *testing.T) {
 	t.Parallel()
 	mockT := new(mockTestingT)
 	longSlice := make([]int, 1_000_000)
 	NotSubset(mockT, longSlice, longSlice)
-	Contains(t, mockT.errorString(), `
-	Error Trace:	
-	Error:      	['\x00' '\x00' '\x00'`)
-	Contains(t, mockT.errorString(), `<... truncated> is a subset of ['\x00' '\x00' '\x00'`)
+	Contains(t, mockT.errorString(), "[]int{0, 0, 0")
+	Contains(t, mockT.errorString(), "<... truncated> is a subset of []int{0, 0, 0")
 }
 
 func TestNotSubsetWithMapTooLongToPrint(t *testing.T) {
@@ -4041,10 +4050,8 @@ func TestNotSubsetWithMapTooLongToPrint(t *testing.T) {
 	mockT := new(mockTestingT)
 	longSlice := make([]int, 1_000_000)
 	NotSubset(mockT, map[int][]int{1: longSlice}, map[int][]int{1: longSlice})
-	Contains(t, mockT.errorString(), `
-	Error Trace:	
-	Error:      	map['\x01':['\x00' '\x00' '\x00'`)
-	Contains(t, mockT.errorString(), `<... truncated> is a subset of map['\x01':['\x00' '\x00' '\x00'`)
+	Contains(t, mockT.errorString(), "map[int][]int{1:[]int{0, 0, 0")
+	Contains(t, mockT.errorString(), "<... truncated> is a subset of map[int][]int{1:[]int{0, 0, 0")
 }
 
 func TestSameWithSliceTooLongToPrint(t *testing.T) {

